Mathlib/LinearAlgebra/Eigenspace/Basic.lean

```diff
@@ -58,227 +58,560 @@ open FiniteDimensional Set
 variable {K R : Type v} {V M : Type w} [CommRing R] [AddCommGroup M] [Module R M] [Field K]
   [AddCommGroup V] [Module K V]
 
+/-- Auxiliary copy of `unifEigenspace`. -/
+private def unifEigenspace_aux (f : End R M) (μ : R) (k : ℕ∞) : Submodule R M :=
+  ⨆ l : ℕ, ⨆ _ : l ≤ k, LinearMap.ker ((f - μ • 1) ^ l)
+
+private lemma mem_unifEigenspace_aux {f : End R M} {μ : R} {k : ℕ∞} {x : M} :
+    x ∈ f.unifEigenspace_aux μ k ↔ ∃ l : ℕ, l ≤ k ∧ x ∈ LinearMap.ker ((f - μ • 1) ^ l) := by
+  have : Nonempty {l : ℕ // l ≤ k} := ⟨⟨0, zero_le _⟩⟩
+  rw [unifEigenspace_aux, iSup_subtype', Submodule.mem_iSup_of_directed]
+  simp only [LinearMap.mem_ker, Subtype.exists, exists_prop]
+  intro m n
+  use max m n
+  dsimp
+  constructor <;> apply (f - μ • 1).iterateKer.monotone <;> simp
+
+private lemma unifEigenspace_aux_mono (f : End R M) (μ : R) :
+    Monotone (f.unifEigenspace_aux μ) := by
+  intro k l hkl x
+  rw [mem_unifEigenspace_aux, mem_unifEigenspace_aux]
+  rintro ⟨n, hn, hx⟩
+  use n, hn.trans hkl, hx
```

Comment:
```suggestion
private lemma mem_unifEigenspace_aux {f : End R M} {μ : R} {k : ℕ∞} {x : M} :
    x ∈ f.unifEigenspace_aux μ k ↔ ∃ l : ℕ, l ≤ k ∧ x ∈ LinearMap.ker ((f - μ • 1) ^ l) := by
  have : Nonempty {l : ℕ // l ≤ k} := ⟨⟨0, zero_le _⟩⟩
  rw [unifEigenspace_aux, iSup_subtype', Submodule.mem_iSup_of_directed]
  · simp only [LinearMap.mem_ker, Subtype.exists, exists_prop]
  exact Monotone.directed_le fun m n h => by simpa using (f - μ • 1).iterateKer.monotone h

private lemma unifEigenspace_aux_mono (f : End R M) (μ : R) :
    Monotone (f.unifEigenspace_aux μ) := fun _ _ hkl => biSup_mono fun _ hi => hi.trans hkl
```

I couldn't resist golfing these a bit - entirely optional if you want to use it.